shared-bindings/spitarget/SPITarget.c

```diff
@@ -0,0 +1,182 @@
+    spitarget_spi_target_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_spitarget_spi_target_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(spitarget_spi_target_deinit_obj, spitarget_spi_target_obj_deinit);
+
+//|     def __enter__(self) -> SPITarget:
+//|         """No-op used in Context Managers."""
+//|         ...
+//  Provided by context manager helper.
+
+//|     def __exit__(self) -> None:
+//|         """Automatically deinitializes the hardware on context exit. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+STATIC mp_obj_t spitarget_spi_target_obj___exit__(size_t n_args, const mp_obj_t *args) {
+    mp_check_self(mp_obj_is_type(args[0], &spitarget_spi_target_target_type));
+    spitarget_spi_target_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    common_hal_spitarget_spi_target_deinit(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(spitarget_spi_target___exit___obj, 4, 4, spitarget_spi_target_obj___exit__);
+
+//|     def load_packet(self, mosi_packet: bytearray, miso_packet: bytearray) -> None:
+//|         """Queue data for the next SPI transfer from the main device.
+//|         If a packet has already been queued for this SPI bus but has not yet been transferred, an error will be raised.
+//|
+//|         :param bytearray miso_packet: Packet data to be sent from secondary to main on next request."""
+//|
+STATIC mp_obj_t spitarget_spi_target_load_packet(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    mp_check_self(mp_obj_is_type(pos_args[0], &spitarget_spi_target_type));
+    spitarget_spi_target_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    if (common_hal_spitarget_spi_target_deinited(self)) {
+        raise_deinited_error();
+    }
+    enum { ARG_mosi_packet, ARG_miso_packet };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_mosi_packet,       MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_miso_packet,       MP_ARG_REQUIRED | MP_ARG_OBJ },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    mp_buffer_info_t mosi_bufinfo;
+    mp_get_buffer_raise(args[ARG_mosi_packet].u_obj, &mosi_bufinfo, MP_BUFFER_WRITE);
+
+    mp_buffer_info_t miso_bufinfo;
+    mp_get_buffer_raise(args[ARG_miso_packet].u_obj, &miso_bufinfo, MP_BUFFER_READ);
+
+    if (miso_bufinfo.len != mosi_bufinfo.len) {
+        mp_raise_ValueError(MP_ERROR_TEXT("Packet buffers for an SPI transfer must have the same length."));
+    }
+
+    common_hal_spitarget_spi_target_transfer_start(self, ((uint8_t *)mosi_bufinfo.buf), ((uint8_t *)miso_bufinfo.buf), mosi_bufinfo.len);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(spitarget_spi_target_load_packet_obj, 1, spitarget_spi_target_load_packet);
+
+//|     def try_transfer(self, *, timeout: float = -1) -> bool:
```

Comment:
```suggestion
//|     def wait_transfer(self, *, timeout: float = -1) -> bool:
```

I think I'd call this wait since it doesn't initiate the transfer.

Reply:
Renamed.